### ISSUE TITLE
cert-manager tutorial needs updating

### ISSUE THREAD

**Issue by youngnick** [projectcontour/contour#1570]
The cert-manager tutorial needs a solid update:

- the version of cert-manager is very old and has an issue where it will generate excessive traffic to Let's Encrypt.
- the tutorial uses `j.hept.io` which will break soon if it hasn't already.
- The docs will also need updating after #950, #951, and #1140 are done. This is just to make sure that we don't miss these bits.

**Comment by maurodelazeri** [comment]
it did https://j.hept.io/contour-deployment-rbac 404

**Comment by davecheney** [comment]
Possible duplicate of #1140

**Comment by stevesloka** [comment]
@youngnick you working on this one?

**Comment by youngnick** [comment]
I needed to do some work on it to get my cluster's cert-manager working again. I've got some notes, and will pick this one up once we've burned down the P1s a bit.

**Comment by stevesloka** [comment]
Hey, @vongohren that error is that you're missing proper RBAC permissions for the TLSCertificateDelegation CRD that Contour creates. 

Today, Cert-Manager works with Contour, but you need to use Ingress resources to handle the HTTP01 challenge.

**Comment by youngnick** [comment]
Sorry @vongohren about the delay, I've been working on a new tutorial. But in essence, the way to use cert-manager with IngressRoute (which is now deprecated) or HTTPProxy (the replacement) is to create a dummy Ingress resource that cert-manager can mutate to answer the HTTP01 challenge from Let's Encrypt. Just making sure that I have all my ducks in a row and my config will actually work when I do it from scratch.

**Comment by stevesloka** [comment]
Hey, @vongohren there are some different issues related to your setup that need to be addressed. They are not connected to your primary task of deploying cert-manager:

1. The script you have to deploy Contour's CRDs is missing the spec for `TLSCertificateDelegation`: https://github.com/projectcontour/contour/blob/release-0.10/deployment/common/common.yaml#L151-L163
2. You're running `v0.10.0`, is there a chance you can upgrade?

Once you get Contour running properly (which should just be by adding the `TLSCertificateDelegation` crd which I linked above, I think it should all just work for ya. =) The rest of your config looks good. 

Please reply back with more questions or hit me up in k8s slack `#contour` channel

**Comment by vongohren** [comment]
I will upgrade yes, we are just using common config :)
So I will take these yamls: https://github.com/projectcontour/contour/tree/v0.15.3/examples/common 
make them json and stick into my jsonnet. Then im 100% sure I have the newest version.
The only thing that is specific is that we are adding a loadbalancer IP, but i  guess that is doable in the end like this ```      "type": "LoadBalancer",
      "loadBalancerIP": params.loadBalancerIP,```

So I have the cert manager running, and then when I run up this contour, it should be working? I have not have had teh chance to try it out yet, but just asking a question to be ready for tomorrow.

**Comment by davecheney** [comment]
@vongohren this issue will close very soon. So we don’t loose your question please raise a new issue, feel free to reference this one, but I don’t want to loose the thread of your problem once this issue closes.

**Comment by vongohren** [comment]
@davecheney sure thing!

**Comment by davecheney** [comment]
Thank you. We’re moving quickly to finalise the 1.0 release. I don’t want to loose your question.

**Comment by vongohren** [comment]
@davecheney ☝️ referenced now :)
